### DISCUSSION TITLE
The default keymap is destructive on macOS, please do not replicate the faults of others

### DISCUSSION THREAD

**Discussion by flyx** [zed-industries/zed#7723] [Feature Requests]
macOS, unlike other OSes, uses the Option key to access additional layers of the keyboard. For example, this is the layer when you press Option on the US International layout:

![keyboard-layer](https://github.com/zed-industries/zed/assets/791710/cab6dd10-e51e-4768-ba2f-11d0271be3f7)

There's another layer for Shift+Option.

Hence, pressing any non-modifier key with Option *is already a defined input* in macOS and should not be used in a keymap.

This has been a problem for editors that are not macOS-first for a long time (VSCode etc), but since Zed is macOS-first, it is baffling that it still does that.

This is a problem especially for alternative keyboard layouts like e.g. Neo2, where the Option layer looks like this:

![keyboard-neo](https://github.com/zed-industries/zed/assets/791710/1a5a36d2-39f6-47e1-8429-de22f3e268ba)

So my first experience in Zed was, I wanted to write a colon and the editor instead *deleted the next word*. I think we can agree that this is not good user experience. The default settings should not contradict the OS.

**Comment by mrnugget** [comment]
Thanks for the report! We're aware and there's already and issue here: https://github.com/zed-industries/zed/issues/7688

The big question is: do we make it easier to disable these keybindings, or do we change all the default keybindings (breaking change)?

**Reply by flyx** [reply]
Since there is already support for boolean expressions in `"context":`, I think an obvious solution would be to move the problematic bindings to a new section, e.g.

```js
{
  "context": "Editor && !macAffinity",
  "bindings": /*...*/
}
```

Then you can introduce new macOS-specific bindings in an alternative section with `Editor & macAffinity` and ask the user on installation or update whether they want macOS-affine keybindings.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
